src/wrapper-utils.js

```diff
@@ -0,0 +1,191 @@
+ * @param {*} newFn
+ * @param {*} origFn
+ */
+function wrapToString (newFn, origFn) {
+    if (!newFn || !origFn) {
+        return
+    }
+    newFn.toString = function () {
+        if (this === newFn) {
+            return functionToString.call(origFn)
+        } else {
+            return functionToString.call(this)
+        }
+    }
+}
+
+function getObjectFromPath (objPath, base = globalObj) {
+    const path = objPath.split('.')
+    if (path.length === 0) {
+        throw new Error('Invalid object path')
+    }
+
+    let object = base
+    for (const pathPart of path) {
+        if (!(pathPart in object)) {
+            // this happens if the object is not implemented in the browser
+            return
+        }
+        object = object[pathPart]
+    }
+    return object
+}
+
+/**
+ * Wrap a get/set or value property descriptor. Only for data properties. For methods, use wrapMethod(). For constructors, use wrapConstructor().
+ * @param {any} object - object or string object path within the global object
+ * @param {string} propertyName
+ * @param {Partial<PropertyDescriptor>} descriptor
+ * @returns {PropertyDescriptor|undefined} original property descriptor, or undefined if it's not found
+ */
+export function wrapProperty (object, propertyName, descriptor) {
+    if (typeof object === 'string') {
+        object = getObjectFromPath(object)
+    }
+    const origDescriptor = getOwnPropertyDescriptor(object, propertyName)
+    if (!origDescriptor) {
+        // this happens if the property is not implemented in the browser
+        return
+    }
+
+    if (('value' in origDescriptor && 'value' in descriptor) ||
+        ('get' in origDescriptor && 'get' in descriptor) ||
+        ('set' in origDescriptor && 'set' in descriptor)
+    ) {
+        wrapToString(descriptor.value, origDescriptor.value)
+        wrapToString(descriptor.get, origDescriptor.get)
+        wrapToString(descriptor.set, origDescriptor.set)
+
+        defineProperty(object, propertyName, {
+            ...origDescriptor,
+            ...descriptor
+        })
+        return origDescriptor
+    } else {
+        // if the property is defined with get/set it must be wrapped with a get/set. If it's defined with a `value`, it must be wrapped with a `value`
+        throw new Error(`Property descriptor for ${propertyName} may only include the following keys: ${objectKeys(origDescriptor)}`)
+    }
+}
+
+/**
+ * Wrap a method descriptor. Only for function properties. For data properties, use wrapProperty(). For constructors, use wrapConstructor().
+ * @param {any} object - object or string object path within the global object
+ * @param {string} propertyName
+ * @param {(originalFn, ...args) => any } wrapperFn - wrapper function receives the original function as the first argument
+ * @returns {PropertyDescriptor|undefined} original property descriptor, or undefined if it's not found
+ */
+export function wrapMethod (object, propertyName, wrapperFn) {
```

Comment:
This method and wrapProperty all dead code as it stands. I would keep it in the PR that needs it.